### DISCUSSION TITLE
Drag-and-drop support

### DISCUSSION THREAD

**Discussion by j-mir-prazak** [fat-tire/resolve#20] [Ideas]
I have drag-and-drop working on Gnome (i suspect it might work the same way on other DE). If you mount host /media to container /media and host /home to container /home, you can do drag-and-drop this way (if you do d&d over terminal, you can see that it drops the address of the file to the terminal - hence the idea).

it might prove messy, but if you cannot work without it, it can help. or at least it might be starting point...

**Comment by fat-tire** [comment]
This is a great solution if you need to drag/drop- but note that doing so would give access to anything on your `/media` and `/home` directories on the host to the container-- which would lose some of (but not all) of the advantages of isolation. I wonder if there's a way to do it more securely (?)

If you physically drag/drop a file, it seems such an explicit action should give permission to the container to access the file.  Android has a similar mechanism for granting file access.  I wonder if an on-the-fly mount/mapping is supported in Podman or Docker, where the action of drag/dropping (or choosing a file or folder from a picker) grants access to the container to access those files/folders..

Does such a thing exist?  I'm fairly new to containers so I'm not sure what's possible.

Incidentally, if someone wants to not worry about isolation and just grant folders access as you suggest-- you can set this up using a `.resolve.rc` file so that it's automatic.

First edit an `.rc` file somewhere:

`gedit ${HOME}/.resolve.rc`  (or however/wherever you want to create this file)

Then add something like this:

```
# .resolve.rc
# Enable drag/drop in /home & /media
RESOLVE_BIND_SOURCES=("/media", "/home")
RESOLVE_BIND_TARGETS=("/media", "/home")
```

Run resolve like:

`RESOLVE_RC_PATH=~/.resolve.rc resolve.sh`

(in the `~/.local/share/applications/resolve.desktop` file, you'd change the exec line to look like this)

`Exec=bash -c "cd $HOME/podman/resolve && env RESOLVE_RC_PATH=$HOME/.resolve.rc ./resolve.sh"`

You can also put that `RESOLVE_RC_PATH` line in `.bashrc` or `.zshrc` or however you like to set environment variables.

Let me know if this works as a more "durable" solution!
